recovery_backend/server/router.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher/api"
+	"github.com/rancher/go-rancher/client"
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type HandleFuncWithError func(http.ResponseWriter, *http.Request) error
+
+const (
+	RetryCounts   = 5
+	RetryInterval = 100 * time.Millisecond
+)
+
+func HandleError(s *client.Schemas, t HandleFuncWithError) http.Handler {
```

Comment:
No need to export because it is just used in NewRouter.

P.S. the code seems repeated, so suggest reusing the code via common lib.

Reply:
Updated. Thank you.